### PR TITLE
Bump etcd from v3.5.10 to v3.5.12

### PR DESCRIPTION
**Pull request by bzsuni**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
https://github.com/etcd-io/etcd/compare/v3.5.10...v3.5.12
#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
none
```

### CODE REVIEW

File: hack/unwanted-dependencies.json [hunk 1]
```diff
@@ -218,7 +218,6 @@
         "go.etcd.io/etcd/api/v3",
         "go.etcd.io/etcd/client/v3",
         "go.etcd.io/etcd/server/v3",
-        "go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc",
```
Comment (pacoxu) [inline hack/unwanted-dependencies.json:221]: 👍

**Comment by pacoxu** [comment]
> /lgtm

/lgtm cancel
The integration test failure is caused by the bump.

File: test/integration/apiserver/tracing/tracing_test.go [hunk 2]
```diff
@@ -338,7 +338,7 @@ endpoint: %s`, listener.Addr().String())), os.FileMode(0755)); err != nil {
 							return v.GetStringValue() != ""
 						},
 						"client": func(v *commonv1.AnyValue) bool {
-							return v.GetStringValue() == "127.0.0.1"
+							return v.GetStringValue() == "192.168.0.1"
```
Comment (ahrtr) [inline test/integration/apiserver/tracing/tracing_test.go:341]: Out of curiosity, is this related to bumping etcd sdk version?

Comment (bzsuni) [reply test/integration/apiserver/tracing/tracing_test.go]: No, it's not related to bumping the etcd SDK version. I added that code snippet just for debugging purposes. I'll remove it shortly.
Thank you for bringing it to my attention.

**Comment by bzsuni** [comment]
> /retest

I've encountered a roadblock where the CI is throwing an error, and I'm unsure how to resolve it. Do you have any suggestions or ideas on how to tackle this issue? @ahrtr

**Comment by ahrtr** [comment]
> I've encountered a roadblock where the CI is throwing an error, and I'm unsure how to resolve it. Do you have any suggestions or ideas on how to tackle this issue? @ahrtr

I am not familiar with the test case either, but I will try to take a look after the CI finishes.

**Comment by ahrtr** [comment]
Confirmed that the CI failure is caused by this PR.  Running the same test on master passed, but failed on this PR.

I see that there is a breaking change in [OpenTelemetry 1.20 ](https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.20.0), but based on the error message (see below), it seems it has nothing to do the otel version (cc otel maintainer @pellared to double confirm, thx). Instead it's a network issue. cc network expert @aojea , thx

Another suspect is grpc-go. etcd 3.5.12 depends on **grpc-go 1.59.0**, but I do not see the release in [grpc-go's release page](https://github.com/grpc/grpc-go/releases). I can only see 1.58.x and 1.60.x.  cc @dfawley and @zasweq for options. thx

```
{Failed;Failed;Failed;Failed;Failed;Failed;Failed;  === RUN   TestAPIServerTracing/create_node
W0209 10:24:57.570395  107924 lease.go:265] Resetting endpoints for master service "kubernetes" to [127.0.0.1]
E0209 10:24:57.571739  107924 controller.go:159] unable to sync kubernetes service: Endpoints "kubernetes" is invalid: subsets[0].addresses[0].ip: Invalid value: "127.0.0.1": may not be in the loopback range (127.0.0.0/8, ::1/128)
W0209 10:25:07.571063  107924 lease.go:265] Resetting endpoints for master service "kubernetes" to [127.0.0.1]
E0209 10:25:07.572432  107924 controller.go:159] unable to sync kubernetes service: Endpoints "kubernetes" is invalid: subsets[0].addresses[0].ip: Invalid value: "127.0.0.1": may not be in the loopback range (127.0.0.0/8, ::1/128)
W0209 10:25:17.571577  107924 lease.go:265] Resetting endpoints for master service "kubernetes" to [127.0.0.1]
E0209 10:25:17.573022  107924 controller.go:159] unable to sync kubernetes service: Endpoints "kubernetes" is invalid: subsets[0].addresses[0].ip: Invalid value: "127.0.0.1": may not be in the loopback range (127.0.0.0/8, ::1/128)
    tracing_test.go:1003: Timed out waiting for trace
    --- FAIL: TestAPIServerTracing/create_node (30.00s)

```

**Comment by dfawley** [comment]
The release shows up for me.  Github has a bug or something with showing the releases out of order sometimes.

https://github.com/grpc/grpc-go/releases/tag/v1.59.0

As far as your test failure, I don't see much to go on, but you could try replacing the grpc dependency with the older version that was used by etcd 3.5.10 to determine if it could be the cause.

**Comment by ahrtr** [comment]
Thanks @dfawley for the feedback.

@bzsuni could you please try to downgrade the dependencies' versions to narrow down the reason? 
- Downgrade grpc-go's version to 1.58.3
- Downgrade otel's version to 1.19.0

If it works, then please try to only downgrade one of them.

**Comment by bzsuni** [comment]
Is it the following operation? @ahrtr 
```
go mod edit -require google.golang.org/grpc@v1.58.3
hack/update-vendor.sh
```

**Comment by ahrtr** [comment]
Firstly you need to run the following command for each go module,
```
go get google.golang.org/grpc@v1.58.3
go mod tidy
```

Secondly, re-vendor the dependencies. I am not sure whether `hack/update-vendor.sh` is the right script. Is it the script that you used when creating this PR? also cc @dims @liggitt to double confirm.

**Comment by pellared** [comment]
Fyi I am not available for at least a week

**Comment by bzsuni** [comment]
Is this issue similar to #112388？
@pohly

**Comment by pohly** [comment]
https://github.com/kubernetes/kubernetes/issues/112388 is just dealing with the same error message. That PR is unrelated.

**Comment by jmhbnz** [comment]
Closing as we have now merged update to `3.5.13` in https://github.com/kubernetes/kubernetes/pull/124027.

/close
